testsuite/model/src/test/java/org/keycloak/testsuite/model/KeycloakModelTest.java

```diff
@@ -550,7 +550,7 @@ protected <R> R inComittedTransaction(Function<KeycloakSession, R> what) {
 
     protected <T, R> R inComittedTransaction(T parameter, BiFunction<KeycloakSession, T, R> what, BiConsumer<KeycloakSession, T> onCommit, BiConsumer<KeycloakSession, T> onRollback) {
         AtomicReference<R> res = new AtomicReference<>();
-        KeycloakModelUtils.runJobInTransaction(getFactory(), session -> {
+        KeycloakModelUtils.runJobInRetriableTransaction(getFactory(), session -> {
```

Comment:
Could you please make this behaviour configurable via a system property?
This should not be applied across all transactions for all databases like it is done here.
This can be done for likes of CRDB but has to be configurable, and by default use the `runJobInTransaction`.

Reply:
Changed as requested, and tested locally by debugging the test. Will ask for another review once the builds are green.